libraries/AC_PrecLand/AC_PrecLand_Companion.cpp

```diff
@@ -46,9 +46,19 @@ void AC_PrecLand_Companion::handle_msg(const mavlink_landing_target_t &packet, u
 {
     _distance_to_target = packet.distance;
 
-    // compute unit vector towards target
-    _los_meas_body = Vector3f(-tanf(packet.angle_y), tanf(packet.angle_x), 1.0f);
-    _los_meas_body /= _los_meas_body.length();
+    if (packet.position_valid == 1) {
+        if (packet.frame == MAV_FRAME_BODY_FRD) {
+            _los_meas_body = Vector3f(packet.x, packet.y, packet.z);
+            _los_meas_body /= _distance_to_target;
+        } else {
+            //we do not support this frame
```

Comment:
single use text msg for when we get a frame we don't understand

Reply:
Hi @tridge Thank you for reviewing. I am afraid that I do not understand what you mean. Would you be able to explain further?

Reply:
@chobitsfan the problem is that if someone sends in a frame other than MAV_FRAME_BODY_FRD, precision landing won't work and the user will have no clue why... 
We should provide some sort of message to the user in this case.. like a single-use text message. For example: send: `send_text(MAV_SEVERITY_INFO,"Plnd: Frame not supported ");` once

Reply:
> @chobitsfan the problem is that if someone sends in a frame other than MAV_FRAME_BODY_FRD, precision landing won't work and the user will have no clue why... We should provide some sort of message to the user in this case.. like a single-use text message. For example: send: `send_text(MAV_SEVERITY_INFO,"Plnd: Frame not supported ");` once
But it main question "is it needs to be ended once?" Or it can be repeated at intervals? Becouse some times you are connected to drone after some time - and message will be available on logs only.

Reply:
Hi @rishabsingh3003 Thank you for explaining. I understand it now.